wandb/compat/tempfile.py

```diff
@@ -10,74 +10,83 @@
 import warnings as _warnings
 from shutil import rmtree as _rmtree
 
-from .weakref import finalize
-
```

Comment:
I had thought too many open file woes were caused by this sketchy implementation so I tried to ensure we used the system TemporaryDirectory for anything but py2.